packages/amplify-graphql-relational-transformer/src/has-many/has-many-directive-sql-transformer.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  ModelDataSourceStrategySqlDbType,
+  TransformerContextProvider,
+  TransformerPrepareStepContextProvider,
+  TransformerTransformSchemaStepContextProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { setFieldMappingResolverReference } from '../resolvers';
+import { HasManyDirectiveConfiguration } from '../types';
+import { validateParentReferencesFields, ensureReferencesArray, getReferencesNodes } from '../utils';
+import { DataSourceBasedDirectiveTransformer } from '../data-source-based-directive-transformer';
+
+/**
+ * HasManyDirectiveSQLTransformer executes transformations based on `@hasMany(references: [String!])` configurations
+ * and surrounding TransformerContextProviders for SQL data sources.
+ */
+export class HasManyDirectiveSQLTransformer implements DataSourceBasedDirectiveTransformer<HasManyDirectiveConfiguration> {
+  dbType: ModelDataSourceStrategySqlDbType;
+  constructor(dbType: ModelDataSourceStrategySqlDbType) {
+    this.dbType = dbType;
+  }
+
+  prepare = (context: TransformerPrepareStepContextProvider, config: HasManyDirectiveConfiguration): void => {
+    const modelName = config.object.name.value;
+    setFieldMappingResolverReference(context, config.relatedType?.name?.value, modelName, config.field.name.value, true);
+  };
+
+  transformSchema = (context: TransformerTransformSchemaStepContextProvider, config: HasManyDirectiveConfiguration): void => {
+    validateParentReferencesFields(config, context as TransformerContextProvider);
+  };
+
+  /** no-op */
+  generateResolvers = (_context: TransformerContextProvider, _config: HasManyDirectiveConfiguration): void => {
```

Comment:
nit: Is the `_` added in the param due to the no op of this function?

Reply:
It is, yes. Is there a better way to do this?